module/actor/data/character.ts

```diff
@@ -0,0 +1,1572 @@
+    dodge: { bonus: 0 },
+  }
+
+  equippedparry: number = 0
+  equippedblock: number = 0
+  currentdodge: number = 0
+  currentmove: number = 0
+  currentsprint: number = 0
+  currentflight: number = 0
+
+  moveoverride: { maneuver: string | null; posture: string | null } = {
+    maneuver: null,
+    posture: null,
+  }
+
+  protected _globalBonuses: AnyObject[] = []
+
+  /* ---------------------------------------- */
+  /*  Data Preparation                        */
+  /* ---------------------------------------- */
+
+  /**
+   * Prepare data related to this DataModel itself, before any derived data is computed.
+   */
+  override prepareBaseData() {
+    super.prepareBaseData()
+
+    // Reset the conditions object
+    // TODO: verify whether this should be fully reset
+    this.conditions = {
+      ...this.conditions,
+      posture: 'standing',
+      maneuver: 'ready',
+      self: { modifiers: [] },
+      target: { modifiers: [] },
+      usermods: new Set<string>(),
+      reeling: false,
+      exhausted: false,
+    }
+
+    // Add the default size modifier to the target conditions
+    if (this.traits.sizemod !== 0) {
+      const sizeModifier = this.traits.sizemod > 0 ? `+${this.traits.sizemod}` : `${this.traits.sizemod}`
+      this.conditions.target?.modifiers?.push(game.i18n?.format('GURPS.modifiersSize', { sm: sizeModifier }) ?? '')
+    }
+
+    // Reset the hitlocationNames object
+    this.hitlocationNames = this.#prepareHitLocationNames()
+
+    // Reset the calculated values of attributes
+    Object.keys(this.attributes).forEach(key => {
+      const attribute = this.attributes[key as keyof typeof this.attributes]
+      this.attributes[key as keyof typeof this.attributes].value = attribute.import
+    })
+  }
+
+  /* ---------------------------------------- */
+
+  override prepareEmbeddedDocuments(): void {
+    this._globalBonuses = this.parent.items.reduce((acc: AnyObject[], item) => {
+      if (!(item.system instanceof BaseItemModel)) return acc
+      acc.push(...item.system.getGlobalBonuses())
+      return acc
+    }, [])
+
+    for (const item of this.parent.items) {
+      if (!(item.system instanceof BaseItemModel)) continue
+      item.system.applyBonuses(this._globalBonuses)
+    }
+
+    this.ads = this.parent.items.filter(item => item.isOfType('feature'))
+    this.skills = this.parent.items.filter(item => item.isOfType('skill'))
+    this.spells = this.parent.items.filter(item => item.isOfType('spell'))
+    const equipment = this.parent.items.filter(item => item.isOfType('equipment'))
+    this.allEquipment = equipment
+    this.equipment = {
+      carried: equipment.filter(item => item.system.carried === true),
+      other: equipment.filter(item => item.system.carried === false),
+    }
+
+    this.melee = this.parent.getItemAttacks({ attackType: 'melee' })
+    this.ranged = this.parent.getItemAttacks({ attackType: 'ranged' })
+
+    this.reactions = this.parent.getItemReactions('reactions')
```

Comment:
I find these two lines mildly surprising. I don't think GurpsActorV2 should be responsible for setting up the data for Character. In fact, I think that Character should not know about its parent at all, other than its some form of Actor. This sets up a circular dependency.

Reply:
Makes sense, I'll get that changed.